inference-engine/src/cldnn_engine/ops/convolution.cpp

```diff
@@ -303,6 +303,78 @@ static void CreateDeformableConvolutionOp(Program& p, const std::shared_ptr<ngra
     }
 }
 
+static void CreateDeformableConvolutionOp(Program& p, const std::shared_ptr<ngraph::op::v8::DeformableConvolution>& op) {
+    p.ValidateInputs(op, {3, 4});
+    auto inputs = p.GetInputPrimitiveIDs(op);
+    std::string layerName = layer_type_name_ID(op);
+
+    auto params = GetConvolutionParameters(op->get_pads_begin(), op->get_dilations(), op->get_strides(), op->get_group());
+    auto outDims = op->get_output_shape(0);
+
+    std::vector<cldnn::primitive_id> weights = {inputs[2]};
+    // Remove weights from inputs
+    inputs.erase(inputs.begin() + 2);
+    if (params.groups == 1 && op->get_input_element_type(0).bitwidth() == 32) {
```

Comment:
Why do we have such limitations for `bitwidth`? Does it really work let's say for `int32`? Otherwise it should be comparation with something like `ngraph::element::Type_t::f32`

Reply:
The original implementation failed on some of SLTs I've added in fp16 precision. I drilled down into the problem and figured out that it's related to fp16 precision math in the kernel, which was not enough for deformable offsets calculation, so that neighbor feature value was calculated instead the correct one. I fixed the problem adding casting of deformable offset value to fp32 in the new patch set, therefore the limitation for optimized deformable convolution implementation was eliminated.